soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -658,42 +598,124 @@ void CheckTrackerItemReceive(GetItemEntry giEntry) {
+                } else if (flag == INFTABLE_193) {
+                    SetCheckCollected(RC_LW_DEKU_SCRUB_GROTTO_FRONT);
+                }
+            }
+            break;
+        case FLAG_ITEM_GET_INF:
+            if (!IS_RANDO) {
+                if (flag == ITEMGETINF_OBTAINED_STICK_UPGRADE_FROM_STAGE) {
+                    SetCheckCollected(RC_DEKU_THEATER_SKULL_MASK);
+                    return;
+                } else if (flag == ITEMGETINF_OBTAINED_NUT_UPGRADE_FROM_STAGE) {
+                    SetCheckCollected(RC_DEKU_THEATER_MASK_OF_TRUTH);
+                    return;
+                } else if (flag == ITEMGETINF_0B) {
+                    SetCheckCollected(RC_HF_DEKU_SCRUB_GROTTO);
+                    return;
+                }
+            }
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_ITEM_GET_INF;
+            break;
+        case FLAG_RANDOMIZER_INF:
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_RANDOMIZER_INF;
+            break;
     }
-    if (gPlayState->msgCtx.msgMode != MSGMODE_NONE) {
-        checkAreas.push_back(checkArea);
-        messageCloseCheck = true;
+    if (checkMatchType == SpoilerCollectionCheckType::SPOILER_CHK_NONE) {
         return;
     }
-    if (IS_RANDO || (!IS_RANDO && giEntry.getItemCategory != ITEM_CATEGORY_JUNK)) {
-        checkAreas.push_back(checkArea);
-        checkCollected = true;
+    for (auto [rc, rcObj] : RandomizerCheckObjects::GetAllRCObjects()) {
+        if ((!IS_RANDO && ((rcObj.vOrMQ == RCVORMQ_MQ && !IS_MASTER_QUEST) ||
+                           (rcObj.vOrMQ == RCVORMQ_VANILLA && IS_MASTER_QUEST))) ||
+            (IS_RANDO && ((OTRGlobals::Instance->gRandomizer->masterQuestDungeons.contains(rcObj.sceneId) &&
+                                rcObj.vOrMQ == RCVORMQ_VANILLA) ||
+                          !OTRGlobals::Instance->gRandomizer->masterQuestDungeons.contains(rcObj.sceneId) &&
+                                rcObj.vOrMQ == RCVORMQ_MQ))) {
+            continue;
+        }
+        SpoilerCollectionCheck scCheck = Location(rc)->GetCollectionCheck();
+        SpoilerCollectionCheckType scCheckType = scCheck.type;
+        if (checkMatchType == SpoilerCollectionCheckType::SPOILER_CHK_RANDOMIZER_INF &&
+              (scCheckType == SpoilerCollectionCheckType::SPOILER_CHK_MERCHANT ||
+               scCheckType == SpoilerCollectionCheckType::SPOILER_CHK_SHOP_ITEM ||
+               scCheckType == SpoilerCollectionCheckType::SPOILER_CHK_COW ||
+               scCheckType == SpoilerCollectionCheckType::SPOILER_CHK_SCRUB ||
+               scCheckType == SpoilerCollectionCheckType::SPOILER_CHK_MASTER_SWORD ||
+               scCheckType == SpoilerCollectionCheckType::SPOILER_CHK_RANDOMIZER_INF)) {
+            if (flag == OTRGlobals::Instance->gRandomizer->GetRandomizerInfFromCheck(rc)) {
+                SetCheckCollected(rc);
+                return;
+            }
```

Comment:
Another doesn't need to happen now, but all of these should get migrated to just using SPOILER_CHK_RANDOMIZER_INF with their associated flags, then we can skip the `GetRandomizerInfFromCheck` check.